hooks/gate-scripts/lib/marker_check.py

```
@@ -2089,6 +2089,219 @@ def _glob_helper_targeted(word, deep=None):
 
 
+# Whole-token POSIX/bash digit-negation globs (#776). Optional stars either side.
+_DIGIT_NEGATION_ONLY_RE = re.compile(r"^\*?\[(?:!|\^)0-9\]\*?$")
+
+
+def _is_digit_negation_only_segment(seg) -> bool:
+    """True when `seg` is only a digit-negation glob (POSIX numeric-validation pattern)."""
+    return bool(_DIGIT_NEGATION_ONLY_RE.match(seg.strip()))
+
+
+def _case_pattern_list_span(pairs, i):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_case_pattern_list_span`

(ANN202)